### PR TITLE
Add sample code of  GestureDetector with no children

### LINKED ISSUE #41962: Add sample of widget containing `gestureDetector` in its documentation
In the [documentation for the gestureDetector](https://github.com/flutter/flutter/blob/2d2a1ffec9/packages/flutter/lib/src/widgets/gesture_detector.dart#L151) Please add another sample with a widget containing a gestureDetector as a child with no children. 

Change the title of the original sample to something like "Sample code of a colored box wrapped in the `gestureDetector` widget"  and the remark inside to reflect that. Something like: 

    // Sample code of a colored box wrapped in gestureDetector 
    // which changes its color when tapped upon. 

The second sample's heading should be something like: "Sample code of a colored box containing a `gestureDetector` widget....

Or just add the "deubg" icon to the [documentation page](https://api.flutter.dev/flutter/widgets/GestureDetector-class.html) on api.flutter.dev so that I can do it myself. 

Or do both... (add debug capability to the doc page, and also add the second sample)

### PR DESCRIPTION
**Pull request by Hattomo**
## Description
This adding a sample code for the GestureDetector widget.
Close #41962

## Related Issues

#41962
Closed PR #43786

## Tests
This is just adding document , so None

## Checklist
Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/gesture_detector.dart [hunk 1]
```diff
@@ -166,6 +168,27 @@ class GestureRecognizerFactoryWithHandlers<T extends GestureRecognizer> extends
...
 ///
+/// {@tool snippet}
+///
+/// This example of a [Container] wraps a [GestureDetector] widget.
+/// Since the [GestureDetector] does not have a child it takes on the size of
+/// its parent making the entire area of the surrounding [Container] clickable.
+/// When tapped the [Container] turns yellow by setting the `_color` field:
+///
+/// ```dart
+/// Container(
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/gesture_detector.dart:179]: Can we specify a size for the Container to make this example a little clearer? Or maybe use a SizedBox?

**Review by Piinks — COMMENTED** [review]
Hi @Hattomo thanks for the contribution!

File: packages/flutter/lib/src/widgets/gesture_detector.dart (hunk 1 above)
Comment (Hattomo) [reply packages/flutter/lib/src/widgets/gesture_detector.dart:179]: Done

**Review by Piinks — COMMENTED** [review]
It looks like this branch may need to be updated with changes from master in order for the checks to pass. I believe the failure is unrelated to your change.

File: packages/flutter/lib/src/widgets/gesture_detector.dart [hunk 2]
```diff
@@ -166,6 +168,28 @@ class GestureRecognizerFactoryWithHandlers<T extends GestureRecognizer> extends
...
+///   color: _color,
+///   height: 200.0,
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/gesture_detector.dart:181]: This should probably also have a width if we specify a height.

Comment (Hattomo) [reply packages/flutter/lib/src/widgets/gesture_detector.dart:181]: Done✔

**Review by Piinks — APPROVED** [review]
LGTM! Thank you for the contribution! :)

**Pull request merged**